src/main/java/it/pagopa/transactions/commands/handlers/TransactionUserCancelHandler.java

```diff
@@ -0,0 +1,86 @@
+package it.pagopa.transactions.commands.handlers;
+
+import com.azure.core.util.BinaryData;
+import com.azure.storage.queue.QueueAsyncClient;
+import it.pagopa.ecommerce.commons.documents.v1.TransactionUserCanceledEvent;
+import it.pagopa.ecommerce.commons.domain.v1.Transaction;
+import it.pagopa.ecommerce.commons.domain.v1.pojos.BaseTransaction;
+import it.pagopa.ecommerce.commons.domain.v1.pojos.BaseTransactionExpired;
+import it.pagopa.ecommerce.commons.domain.v1.pojos.BaseTransactionWithPaymentToken;
+import it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto;
+import it.pagopa.transactions.commands.TransactionUserCancelCommand;
+import it.pagopa.transactions.exceptions.AlreadyProcessedException;
+import it.pagopa.transactions.repositories.TransactionsEventStoreRepository;
+import it.pagopa.transactions.utils.TransactionsUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+@Component
+@Slf4j
+public class TransactionUserCancelHandler implements
+        CommandHandler<TransactionUserCancelCommand, Mono<TransactionUserCanceledEvent>> {
+
+    private final TransactionsEventStoreRepository<Void> transactionEventUserCancelStoreRepository;
+    private final QueueAsyncClient transactionClosureQueueAsyncClient;
+
+    private final TransactionsUtils transactionsUtils;
+
+    @Autowired
+    public TransactionUserCancelHandler(
+            TransactionsEventStoreRepository<Void> transactionEventUserCancelStoreRepository,
+            @Qualifier("transactionClosureQueueAsyncClient") QueueAsyncClient transactionClosureQueueAsyncClient,
+            TransactionsUtils transactionsUtils
+    ) {
+        this.transactionsUtils = transactionsUtils;
+        this.transactionEventUserCancelStoreRepository = transactionEventUserCancelStoreRepository;
+        this.transactionClosureQueueAsyncClient = transactionClosureQueueAsyncClient;
+    }
+
+    @Override
+    public Mono<TransactionUserCanceledEvent> handle(TransactionUserCancelCommand command) {
+        Mono<BaseTransaction> transaction = transactionsUtils.reduceEvents(
+                command.getData()
+        );
+
+        return transaction
+                .cast(BaseTransaction.class)
+                .onErrorMap(err -> new AlreadyProcessedException(command.getData()))
```

Comment:
Remove this since if no transaction is found for the input transactionId a Mono.error with TransactionNotFoundException is already returned by `transactionsUtils.reduceEvents` method.
Beside that AlreadyProcessedException is mapped to 409 http response code that is wrong in this case